src/item_factory.cpp

```diff
@@ -3121,6 +3121,22 @@ bool Item_factory::load_string( std::vector<std::string> &vec, const JsonObject
     return result;
 }
 
+namespace
+{
+auto load_active( std::vector<ItemFn> &xs, const JsonObject &obj ) -> bool
+{
+    const bool result = obj.has_bool( "active" ) && obj.get_bool( "active" );
+    if( result ) {
+        xs.emplace_back( []( detached_ptr<item> &&it ) {
+            it->activate();
+            return it;
```

Comment:
GCC will select the copy constructor when returning arguments which is deleted for detached pointers. std::move should give it the nudge it needs to select the move constructor. To appease GCC if you're returning an argument detached pointer you need to move it. The reverse is true for local variable detached pointers, you can't move them or MSVC will fail. 
```suggestion
            return std::move(it);
```

Reply:
thanks! do you think it'd make sense to migrate other engine apis like this in future?